tests/test_git.py

```diff
@@ -283,15 +283,25 @@ def test_clone_brings_branches(self):
 
     def test_list_branches_detached_head_state(self):
         tf1 = tempfile.NamedTemporaryFile(dir=self.repo.repo_dir)
-        out = self.repo.run_command(
+        out1 = self.repo.run_command(
             ['git', 'stash', 'save', '-u', 'stash message'])
-        self.repo.checkout('stash@{0}')
+        out2 = self.repo.checkout('stash@{0}')
         branches = self.repo.list_branches()
-        self.assertIn({
-            'commit': self.repo.head,
-            'message': 'On master: stash message',
-            'branch': '(detached from %s)' % self.repo.head[:7]
-        }, branches)
+        selected = None
+        for b in branches:
+            if b['commit'] == self.repo.head:
+                selected = b
+                break
```

Comment:
No changes required, just musings on python...

I wonder/wish if there was a simpler pattern for this in python. Its such a common occurrence.

This works but its not inconclusively simpler or clearer.
```python
        selected = next((b for b in branches if b['commit'] == self.repo.head), None)
```